### ISSUE TITLE
Update WebDriver to v3.14.0

### ISSUE THREAD

**Issue by itaibh** [appium/appium-dotnet-driver#256]
## Description

When compiling an Appium program with the latest version of WebDriver (v3.14.0) I get warnings about `DesiredCapabilities` class being deprecated, and it seems there is no alternative.
Since I've already switched to the new version, I don't want to go back just so the Appium parts compile without warnings (I test both native programs and web pages using Appium, and test web pages using Selenium WebDriver in the same suite).

## Environment

* .NET client: **ANY**
* Appium server version: v1.8.1
* WebDriver version: v3.14.0

### Web Test

Compiles without warnings

```csharp
var options = new ChromeOptions();
options.PlatformName = "Android";
options.AddAdditionalCapability("deviceName", "Nexus S5");
options.AddAdditionalCapability("platformVersion", "5.1.0");
options.AddAdditionalCapability("newCommandTimeout", 600);
options.AddAdditionalCapability("idleTimeout", 900);
var driver = new RemoteWebDriver(new Uri("http://192.168.57.1:4723/wd/hub/"), options.ToCapabilities());
```

### Native App Test

Warnings about `DesiredCapabilities` class being deprecated

```csharp
var capabilities = new DesiredCapabilities();
capabilities.SetCapability("platformName", "Android");
capabilities.SetCapability("platformVersion", "7.0");
capabilities.SetCapability("deviceName", "Samsung Galaxy S8");

capabilities.SetCapability("app", "http://saucelabs.com/example_files/ContactManager.apk");

var driver = new AndroidDriver<AppiumWebElement>(new Uri("http://127.0.0.1:4723/wd/hub"), capabilities);
```

**Comment by JMelon** [comment]
I am also interested in the answer to this question. Would also like get working example without `DesiredCapabilities`.

**Comment by SomeRandomGuy7** [comment]
Same issue here, would be good to get rid of warnings.

**Comment by akinsolb** [comment]
resolved by #258

**Comment by EricAtHaufe** [comment]
After installing 4.0.0.4 beta, my test code was changed which start desktop application:
                appiumOptions.AddAdditionalCapability("app", @ExecutablePath);
                appiumOptions.AddAdditionalCapability("platformName", "Windows");
                appiumOptions.AddAdditionalCapability("deviceName", "WindowsPC");
                 session = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appiumOptions, TimeSpan.FromSeconds(1.5));

An exception is thrown,  trace states:  Could not parce w3c capabilities: 'deviceName' cann't be blank.  Falling back to JSONWP protocal.

Test case fails since the execption is thrown but my application is launched.   Any idea?
